src/openms/include/OpenMS/METADATA/SpectrumMetaDataLookup.h

```diff
@@ -218,6 +224,30 @@ namespace OpenMS
     }
 
     /**
+     * @brief encapsulates readSpectra with file loading convenience if only a file path is given
+     * @param spectra_file file to the SpectrumContainer, must be valid path to mz file
+     */
+    void readMzFile(const String& spectra_file)
+    {
+      PeakMap exp;
+      FileHandler().loadExperiment(spectra_file, exp, FileTypes::UNKNOWN, ProgressLogger::NONE,
+                                   false, false);
+      this->readSpectra(exp.getSpectra());
+      this->setSpectraData(File::absolutePath(spectra_file));
+    }
+
+    /**
+     * @brief set spectra_data from read SpectrumContainer origin (i.e. filename)
+     *
+     * @param spectra_data the name (and path) of the origin of the read SpectrumContainer
+     */
+    void setSpectraData(const String& spectra_data)
```

Comment:
this name is confusing - particularly in the presence of the readSpectra() function. Better one?